### ISSUE TITLE
IDxcResult output enumeration is reporting multiple DXC_OUT_NONE values and missing supported DXC_OUT_ values.

### ISSUE THREAD

**Issue by coopp** [microsoft/DirectXShaderCompiler#6523]
**Description**
Calling IDxcResult::GetNumOutputs( ) and enumerating outputs by index is returning incorrect results

**Steps to Reproduce**
```c++
IDxcResult* result; // Obtained from Compile( )

uint32_t numOutputs = result->GetNumOutputs();
std::cout << "Compiled result contains " << numOutputs << " outputs\n";
for (uint32_t i = 0; i < numOutputs; ++i)
{
    std::cout << result->GetOutputByIndex(i) << " found\n";
}

```

**Actual Behavior**

**Environment**
- DXC version dxcompiler.dll: 1.8 - 1.8.2403.37 (11e1318c3); dxil.dll: 1.8(101.8.2403.24)
- Host Operating System Windows 11

**Comment by damyanp** [comment]
@coopp - looks like the "Actual Behavior" section needs to be filled in.

**Comment by coopp** [comment]
Actual behavior is filled in.
